lib/collection/src/wal.rs

```diff
@@ -251,6 +240,8 @@ impl<'s, R: DeserializeOwned + Serialize + Debug> SerdeWal<R> {
     ///
     /// If the index is lower than `first_index`, it means we have already acknowledged it but we
     /// are still holding it in a closed segment until it gets truncated.
+    //
+    // TODO: Rename to `first_acknowledged_index`? Or to something else more meaningful than "closed"? 🤔
```

Comment:
`first_acknowledged_index` fine by me, I don't have a strong opinion on either. I do find both names still a bit confusing though.

I choose `closed` at first because we use the same terminology in other places.

Reply:
Decided to ignore this for now, cause I realized what confuses me about the whole `first_index`/`first_closed_index`, is that it _kinda_ makes more sense to give a more specific name to `first_index`, instead of to `first_closed_index`.

Like, `first_closed_index` *is* **the first** index of WAL. And the `first_index` is, like, some semi-arbitrary index somewhere in the WAL. I don't want to dive deeer into it RN, so I leave it as-is. (Just added it to my personal list of Qdrant `TODO`s. :D)

Reply:
Under normal circumstances you'd only use the logical (open) part of the WAL. So in that case, `first_index` probably makes sense. Its only with WAL diff that we want to read over the closed bit too, just because we have that information. Otherwise we'd significantly limit our diffing possibilities.

What about naming it `first_{open,closed}_index()` for now? That may prevent some confusion because you then have to choose a specific method.